policy-controller/k8s/status/src/index.rs

```diff
@@ -114,8 +131,14 @@ impl Controller {
                 biased;
                 res = self.claims.changed() => {
                     res.expect("Claims watch must not be dropped");
-                    tracing::debug!("Lease holder has changed");
                     let claim = self.claims.borrow_and_update();
+                    match (self.leader, claim.is_current_for(&self.name)) {
+                        (true, false) =>
+                            tracing::debug!("Lease holder has changed; no longer the leader"),
+                        (false, true) =>
+                            tracing::debug!("Lease holder has changed; now the leader"),
+                        _ => {}
+                    }
                     self.leader = claim.is_current_for(&self.name);
```

Comment:
```suggestion
                    let was_leader = self.leader;
                    self.leader = claim.is_current_for(&self.name);
                    if was_leader != self.leader {
                        tracing::debug!(leader = %self.leader, "Leadership changed");
                    } else {
                        tracing::trace!(leader = %self.leader, "Leadership unchanged");
                    }
```

Reply:
Even at trace level, logging that leadership is unchanged every 10s does not seem particularly useful.